### ISSUE TITLE
EE 7.0.3 and 7.1: Cannot upload files anymore.

### ISSUE THREAD

**Issue by bytebaby** [ExpressionEngine/ExpressionEngine#2303]
I can no longer upload files (graphics), neither with the file manager nor via the icon in the button bar (text field). EE reports: "The root path /tmp is not readable" (see screenshot) But the file permissions of the directories are correct, the server path is also correct (the affected website has existed since EE 4xx)

![error01](https://user-images.githubusercontent.com/21154749/185595107-82ca2122-1f33-454b-8b10-e4b8f7d73a13.jpeg)

Only when I "synchronize" the picture directory do the files appear in the file manager.

If I then want to crop any graphic, for example, the following error message appears:

![error02](https://user-images.githubusercontent.com/21154749/185595156-ee159a3d-c8e4-4a54-bb6f-f3543ec13cff.jpeg)

I did the basic synchronization in the file manager beforehand.

Tested with PHP 7.4 and 8.1
Browser: Chrome and Firefox

**Comment by intoeetive** [comment]
@bytebaby is this dedicated server / droplet or managed hosting of some kind?

Also, is your system folder above web root?

**Comment by bytebaby** [comment]
No, as always, the EE system directory is under the web root.

**Comment by intoeetive** [comment]
@bytebaby do you have a way to test using file in this PR to see if that solves your issue?
https://github.com/ExpressionEngine/ExpressionEngine/pull/2291/files

**Comment by bytebaby** [comment]
Yes. I've replaced the three snippets in the two files exactly, but I get the following error:

<img width="297" alt="image" src="https://user-images.githubusercontent.com/21154749/185906436-cc2020ff-0e20-4b93-b401-dcc9f2a6da20.png">

The error immediately appears on the homepage of the website.

**Comment by intoeetive** [comment]
Interesting. Can you share some info about your environment?
(OS, PHP version, server path to the site, hosting type, any other useful info...)

**Comment by bytebaby** [comment]
Mac OS 12.5.1
PHP 7.4
/www/htdocs/xxxxxxx/mydoman.tld/mySystemFolder/
Image-Path: /www/htdocs/xxxxxxx/mydomain.tld/images/bilder/
Apache HTTP Server 
MariaDB

I will also test the website locally.

**Comment by appricosa** [comment]
I am experiencing a similar error after upgrading to EE 7 (now 7.1.2). But my error message is a little different:
<img width="962" alt="Bildschirmfoto 2022-08-27 um 11 41 42" src="https://user-images.githubusercontent.com/49977166/187024845-d1f53f25-4019-4d53-8ab8-6cab979da0a2.png">
I had followed the instructions to migrate to the new file manager, synched all files and also created a new upload target. The ~files~ **images** do get uploaded but don't show up in the file manager. Also no **image** manipulations are performed anymore.
This is happening both locally and live.

**Update:** I noticed on live server that EE does not seem to access the server root folder anymore. Instead, it creates the whole folder structure of {base_path} from my website root folder. After uploading the files to that location, of course it cannot find those images inside the defined upload directory.

I tried adding an additional slash before {base_path} but nothing seems to help.

The only solution was to rollback to 6.5.3 for now. File uploads back to normal. Please fix this as it's a very essential feature for a CMS.

In fact I am reconsidering using EE as a CMS at all in the near future with all the flaws each upgrade comes with.

**Comment by bryannielsen** [comment]
@appricosa I definitely understand the frustration and we do apologize for this issue you've encountered when upgrading to EE7.  I realize you've already spent a lot of time upgrading and discovering this issue and then reverting to EE6 so I hate to take up more of your time.  But if it's possible to get a little more information on your setup it may help us reproduce the issue and get a fix for you and other users.

I believe this error is resulting from an issue with creating a temp file which we do during thumbnail creation and other image transformations during an upload.  So my first guess is that permissions may be a problem on your temp folder.  Could you check the path returned by `sys_get_temp_dir()` or this should also be visible in the PHP Info Utility through the control panel at `Tools > Utilities > PHP Info` under `Environment > TEMP` or `$_SERVER['TEMP']`.  And whatever this path is the user running your webserver should have write access to this folder.

If that is not the issue I think filling in the following blanks could us debug a bit further:

```
Temp Path (Found in PHP Info mentioned above):
Default base path (Found in Settings > URL and Path Settings): 
File Upload path (Found in Files > {YOUR_UPLOAD_DIRECTORY} > Edit):
System path (Full path to your system folder): 
Index path (Full path to your index.php file): 
```

**Comment by appricosa** [comment]
Hi Dev Team, Thanks a lot for your valuable feedback.
I have been testing with 6.3.5 and the last updates 7.1.4 and 7.1.5.. so I thought you might hear my experience on this, too.
So with 7.1.4 my upload problem did not get solved right away, but at least the error messages changed to an obvious (read/write access) issue with the PHP Temp folder. Which is strange, because EE 6.3.5 and prior never had any issues creating the temp files. I got my dev environment installed on a Mac and editing the folder permissions down to root level did not work ("operation not allowed" - even with sudo). 

So I have solved this issue by defining a new temp folder via my config file, and Heureka – it worked:

`putenv('TMPDIR=/Applications/XAMPP/xamppfiles/htdocs/renvert/ee/tmp');`

Hope, this is not creating any security or compatibility issues.
Maybe this helps someone else, too.
